uplink/src/collector/downloader.rs

```diff
@@ -378,6 +369,52 @@ pub struct DownloadFile {
     pub download_path: Option<PathBuf>,
 }
 
+// A temporary structure to help us retry downloads
+// that failed after partial completion.
+struct RetryDownloadHandle {
```

Comment:
Why is this called `RetryDownloadHandle`? I don't see any retry code

Reply:
As the comment states, this is used to encapsulate details regarding the download that will be useful in tracking the download and retrying when it fails to complete

Reply:
You don't name structs and methods based on callers. You name them based on what they do

Reply:
Rename this to `DownloadState` and make this member of `FileDownloader`